### PR TITLE
HG-852 Discussion header

### PR DESCRIPTION
**Pull request by mklucsarits**
https://wikia-inc.atlassian.net/browse/HG-852

This implements a header for discussion thread list pages. I've moved re-usable code for Headroom.js to a mixin as part of these changes. Also included here is a change to make the share-feature component show only when explicitly specified by a property.

@lizlux @garthwebb

### CODE REVIEW

**Comment by lizlux** [comment]
Code looks good but I'll re-review when the share header regression is fixed.

File: front/scripts/main/components/ShareFeatureComponent.ts [hunk 1]
```diff
@@ -2,61 +2,19 @@
...
 /// <reference path="../../main/mixins/LanguagesMixin.ts" />
 /// <reference path="../../main/mixins/TrackClickMixin.ts" />
+/// <reference path="../../main/mixins/HeadroomMixin.ts" />
 /// <reference path="../../../../typings/headroom/headroom.d.ts" />
 'use strict';
 
-App.ShareFeatureComponent = Em.Component.extend(App.TrackClickMixin, App.LanguagesMixin, {
+App.ShareFeatureComponent = Em.Component.extend(App.TrackClickMixin, App.LanguagesMixin, App.HeadroomMixin, {
 	classNames: ['share-feature'],
-	tagName: 'div',
```
Comment (mklucsarits) [inline front/scripts/main/components/ShareFeatureComponent.ts]: Removed because it's not needed.

**Comment by lizlux** [comment]
LGTM!

**Comment by kenkouot** [comment]
LGTM

**Pull request merged**